packages/components/src/Navbar/Component.purs

```diff
@@ -66,37 +70,95 @@ navbar = make component
 
 render :: Self -> JSX
 render self@{ props, state } =
-  View.navbar
-    { onLogout:
-        if isJust props.loggedInUser
-        then Just do
-             props.logout
-             send self (CollapsedNavVisibility Hidden)
-        else Nothing
-    , paperInfo
-    , toggleCollapsedNav:
-        send self (CollapsedNavVisibility $ negateVisibility state.collapsedNavVisibility)
-    , collapsedNav:
-        \items ->
-          Collapsed.collapsed
-            { visibility: state.collapsedNavVisibility
-            , navItems: items
-            }
+  DOM.div
+    { className: "nav--navbars"
+    , children:
+        [ fullNav self
+        , hamburgerNav self
+        , collapsedNav self
+        ]
+    }
+
+-- | Full width navbar
+fullNav :: Self -> JSX
+fullNav self@{ props, state } =
+  DOM.div
+    { className: "nav--nav-container"
+    , children:
+        [ paperLogo props.paper
+        , needHelp props.paper
+        , logoutButton self
+        ]
+    }
+
+logoutButton :: Self -> JSX
+logoutButton self = foldMap signOutButton (onLogout self =<< self.props.loggedInUser)
+
+-- | Narrow navbar with hamburger button
+hamburgerNav :: Self -> JSX
+hamburgerNav self@{ props, state } =
+  DOM.div
+    { className: "nav--hamburger-container"
+    , children:
+        [ paperLogo props.paper
+        , hamburgerButton self
+        ]
     }
-  where
-    paperInfo =
-      { logo: paperLogoUrl props.paper
-      , email: paperEmail props.paper
-      , phoneNumber: paperPhoneNumber props.paper
-      }
 
-update :: Self -> Action -> StateUpdate Props State
-update self = case _ of
-  CollapsedNavVisibility v ->
-    Update self.state { collapsedNavVisibility = v }
+-- | Items of hamburger nav
+collapsedNav :: Self -> JSX
+collapsedNav self =
+  Collapsed.collapsed
+    { visibility: self.state.collapsedNavVisibility
+    , navItems: [ logoutButton self, needHelp self.props.paper ]
+    }
+
+onLogout :: Self -> Persona.User -> Maybe (Effect Unit)
+onLogout self _user = Just do
```

Comment:
Let's move the `Maybe` out of here and have the `Just onLogout` at the call sites, if it's actually needed.